src/marks/raster.js

```diff
@@ -375,8 +376,8 @@ function interpolateRandomWalk(index, width, height, X, Y, V) {
       if (x === 0.5) iy = ix;
       let distance; // distance to closest sample
       let step = 0; // count of steps for this walk
-      while ((distance = Math.hypot(X[index[iw]] - cx, Y[index[iw]] - cy)) > 0.5 && step < 2) {
-        const angle = random() * 2 * Math.PI;
+      while ((distance = Math.hypot(X[index[iw]] - cx, Y[index[iw]] - cy)) > 0.5 && step < 3) {
+        const angle = random(x, y, step) * 2 * Math.PI;
```

Comment:
Should we pass cx and cy here instead? Would it look better or worse?

```suggestion
        const angle = random(cx, cy, step) * 2 * Math.PI;
```

Reply:
the quality looks about the same, but we must make sure they're non-negative

Reply:
I think conceptually it makes sense to use x and y because that is the sample we are generating; cx and cy is just an intermediate location. But I don’t really know what I’m talking about… 🤷

Reply:
The noise generator creates a 128x128 texture with "random" numbers. Looking up the value on this texture (with x and y at the first step), neighboring points avoid using the angles that are too similar on their first jump—in a sense, they explore the space better.

Now, we also want two consecutive jumps to go in "well-different" directions—we don't want to continue in a straight line, it would defeat the purpose.

To guarantee that, we could use (cx, cy), when I try it, it seems to work as well. However, cx, cy can be anywhere on the plane, so we'd need to adjust the noise function to allow negative numbers (it's not complicated, but currently only non-negative numbers are accepted, and the solution is not a simple reflection of the positive values with Math.abs). Here instead I use z=step to look up a different area of the texture at each step.